### PR TITLE
chore: upgrade stylelint

### LINKED ISSUE #309: Support stylelint 15
[stylelint 15](https://github.com/stylelint/stylelint/releases/tag/15.0.0) is released, please update the range in `peerDependencies` if the plugin is compatible.

### PR DESCRIPTION
**Pull request by AndyOGo**
fixes #309

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -82,7 +83,7 @@
     "pinst": "^2.1.6",
     "prettier": "^2.2.1",
     "semantic-release": "^17.3.0",
-    "stylelint": "^14.0.1",
+    "stylelint": "^15.0.0",
     "stylelint-test-rule-tape": "^0.2.0",
```
Comment (AndyOGo) [inline package.json:87]: - [ ] [`stylelint-test-rule-tape`](https://github.com/stylelint/stylelint-test-rule-tape) is **archived**

File: package.json [hunk 2]
```diff
@@ -82,7 +83,7 @@
     "pinst": "^2.1.6",
     "prettier": "^2.2.1",
     "semantic-release": "^17.3.0",
-    "stylelint": "^14.0.1",
+    "stylelint": "^15.0.0",
```
Comment (AndyOGo) [inline package.json:86]: Upgrading `v15` yields dozens of type errors
> TypeError: node.rangeBy is not a function
    at Object.report (/Users/andy/work/stylelint-declaration-strict-value/node_modules/stylelint/lib/utils/report.js:35:34)

- [ ] https://github.com/stylelint/stylelint/issues/5756
- [ ] https://github.com/stylelint/stylelint/issues/5766

Comment (AndyOGo) [reply package.json:86]: Seems that some patches are being prepared 🤔 
https://github.com/stylelint/stylelint/issues/6647

File: package.json (hunk 1 above)
Comment (AndyOGo) [reply package.json:87]: tape is replaced by [`jest-preset-stylelint`](https://github.com/stylelint/jest-preset-stylelint)

- https://stylelint.io/developer-guide/plugins/#testing
- https://stylelint.io/developer-guide/system-tests/

File: package.json [hunk 3]
```diff
@@ -63,6 +63,7 @@
     "@semantic-release/changelog": "^5.0.1",
     "@semantic-release/exec": "^5.0.0",
     "@semantic-release/git": "^9.0.0",
+    "@types/stylelint": "^14.0.0",
```
Comment (AndyOGo) [inline package.json:66]: - [ ] Those types are out of date too - https://github.com/stylelint/stylelint/issues/2444

File: package.json (hunk 2 above)
Comment (silverwind) [reply package.json:88]: Does [15.1.0](https://github.com/stylelint/stylelint/releases/tag/15.1.0) help any?

Comment (AndyOGo) [reply package.json:88]: No.
The issue is the old tape test utils, they are deprecate and archived.

The jest test utils do not support testing a plugins options validation.

I will finish it as soon as I free time available.

File: test/custom-message.js [hunk 4]
```diff
@@ -56,9 +50,9 @@ testRule(rule, {
   ],
 });
 
-testRule(rule, {
+testRule({
+  skip: true,
```
Comment (AndyOGo) [inline test/custom-message.js:54]: - [ ] [`jest-preset-stylelint`](https://github.com/stylelint/jest-preset-stylelint) doesn't reject for invalid config.
   https://github.com/stylelint/jest-preset-stylelint/issues/58

File: test/auto-fix-func.js [hunk 5]
```diff
@@ -53,12 +39,15 @@ testRule(
...
+        code: '.foo { color: #fff; }',
+        fixed: '.foo { color: $color-white; }',
+      },
+      {
+        code: '.foo { color: red; }',
+        fixed: '.foo { color: $color-red; }',
+      },
       {
         code: '.foo { font-size: 16px; }',
         message: `"font-size" is not a color property (${ruleName})`,
```
Comment (AndyOGo) [inline test/auto-fix-func.js:53]: - [ ] Custom autofix error message can't be asserted yet.
   https://github.com/stylelint/jest-preset-stylelint/issues/60

**Pull request merged**